frontend/src/core/resource/components/ResourceMenu.js

```diff
@@ -113,19 +158,66 @@ export class ResourceMenuBase extends React.Component<Props, State> {
                         autoFocus
                         value={ this.state.search }
                         onChange={ this.updateResourceList }
+                        placeholder="Filter resources"
+                    />
+                </div>
+
+                <div className="resources-list-header">
+                    <Localized id='resource-ResourceMenu--resource'>
+                        <span className="resource" onClick={ this.toggleResource }>Resource</span>
+                    </Localized>
+                    <span className={"resource icon " + sortResource}
+                          onClick={ this.toggleResource }
+                    />
+                    <Localized id='resource-ResourceMenu--progress'>
+                        <span className="progress" onClick={ this.toggleProgress }>Progress</span>
+                    </Localized>
+                    <span className={"progress icon " + sortProgress}
+                          onClick={ this.toggleProgress }
                     />
                 </div>
 
                 <ul>
-                    { resourceElements.length ?
-                        resourceElements.map((resource, index) => {
+                    { resourceElements.length ? (
+                            (this.state.sortActive === 'resource'
+                                ? resourceElements.sort( (a,b) => {
+                                    let resourceA = getResource(a),
+                                        resourceB = getResource(b);
+
+                                    let result = 0;
+
+                                    if (resourceA < resourceB) {
+                                        result = -1;
+                                    }
+                                    if (resourceA > resourceB) {
+                                        result = 1;
+                                    }
+                                    return this.state.sortAsc ? result :  result * -1;
+
+                                })
+                                : resourceElements.sort( (a,b) => {
```

Comment:
`:` should be alone in this line, `resourceElements...` should go to the next one